fem/gslib.hpp

```diff
@@ -143,39 +145,96 @@ class FindPointsGSLIB
-   const Array<unsigned int> &GetCode() const { return gsl_code; }
+   virtual const Array<unsigned int> &GetCode() const { return gsl_code; }
    /// Return element number for each point found by FindPoints.
-   const Array<unsigned int> &GetElem() const { return gsl_mfem_elem; }
+   virtual const Array<unsigned int> &GetElem() const { return gsl_mfem_elem; }
    /// Return MPI rank on which each point was found by FindPoints.
-   const Array<unsigned int> &GetProc() const { return gsl_proc; }
+   virtual const Array<unsigned int> &GetProc() const { return gsl_proc; }
    /// Return reference coordinates for each point found by FindPoints.
-   const Vector &GetReferencePosition() const { return gsl_mfem_ref;  }
+   virtual const Vector &GetReferencePosition() const { return gsl_mfem_ref;  }
    /// Return distance Distance between the sought and the found point
    /// in physical space, for each point found by FindPoints.
-   const Vector &GetDist()              const { return gsl_dist; }
+   virtual const Vector &GetDist()              const { return gsl_dist; }
 
    /// Return element number for each point found by FindPoints corresponding to
    /// GSLIB mesh. gsl_mfem_elem != gsl_elem for mesh with simplices.
-   const Array<unsigned int> &GetGSLIBElem() const { return gsl_elem; }
+   virtual const Array<unsigned int> &GetGSLIBElem() const { return gsl_elem; }
    /// Return reference coordinates in [-1,1] (internal range in GSLIB) for each
    /// point found by FindPoints.
-   const Vector &GetGSLIBReferencePosition() const { return gsl_ref; }
+   virtual const Vector &GetGSLIBReferencePosition() const { return gsl_ref; }
+};
+
+/** \brief OversetFindPointsGSLIB enables use of findpts for arbitrary number
+ * of overlapping grids. The parameters in this class are the same as
+ * FindPointsGSLIB with the difference of additional inputs required to account
+ * for more than 1 mesh.
+ */
+class OversetFindPointsGSLIB : public FindPointsGSLIB
+{
+protected:
+   bool overset;
+   unsigned int u_meshid;
+   Vector distfint; // Used to store the nodal values of the gridfunction passed
+   // to findpts.
+
+public:
+   OversetFindPointsGSLIB() : FindPointsGSLIB(),
+      overset(true) { }
+
+#ifdef MFEM_USE_MPI
+   OversetFindPointsGSLIB(MPI_Comm _comm) : FindPointsGSLIB(_comm),
+      overset(true) { }
+#endif
+
+   /** Initializes the internal mesh in gslib, by sending the positions of the
+       Gauss-Lobatto nodes of the input Mesh object @a m.
+       Note: not tested with periodic meshes (L2).
+       Note: the input mesh @a m must have Nodes set.
+
+       @param[in] m         Input mesh.
+       @param[in] meshid    A unique # for each overlapping mesh. This id is
+                            used to make sure that points being searched are not
+                            looked for in the mesh that they belong to.
+       @param[in] gfmax     (Optional) GridFunction in H1 that is used as a
```

Comment:
Just for my own edification and this might be a dumb question, how might you define such a gridfunction and would it live on the various meshes? I'm just trying to get a handle for this parameter, since none of the new examples make use of it. The other ones make sense to me at least.

Reply:
I am glad you asked this because I am sure users would have this question. For overlapping Schwarz based calculations, a good option is to use a distance field based on the overlapping boundaries. Picking elements that are as far as possible from the boundaries helps with convergence during Schwarz iterations. I have added some more detail to the comment to clarify.